common/lib/dependabot/config/ignore_condition.rb

```diff
@@ -95,6 +123,45 @@ def ignore_major(version)
 
         ["#{lower_bound}, #{upper_bound}"]
       end
+
+      def ignored_ranges(dependency, security_updates_only)
+        ignored_ranges ||= {}
+        key = "#{dependency.name}-#{dependency.version}-#{security_updates_only}"
+        ignored_ranges[key] ||= if security_updates_only
+                                  versions
+                                elsif versions.empty? && transformed_update_types.empty?
+                                  [ALL_VERSIONS]
+                                else
+                                  versions_by_type(dependency) + versions
+                                end
+      end
+
+      # Custom requirement for ignoring specific git commit SHAs
+      class ShaRequirement < Gem::Requirement
+        def initialize(*requirements)
+          requirements = requirements.flatten.flat_map do |req_string|
+            req_string.split(",").map(&:strip)
+          end
+
+          super(requirements)
+        end
+
+        def self.parse(obj)
+          if obj.is_a?(String) && obj.strip.match?(GIT_SHA_IGNORE_REGEX)
+            return [GIT_SHA_IGNORE_PREFIX, obj.gsub(GIT_SHA_IGNORE_REGEX, "").strip]
+          end
+
+          super
+        end
+
+        def satisfied_by?(version)
+          if requirements.any? { |op, _| op == GIT_SHA_IGNORE_PREFIX }
+            rv = requirements.find { |op, _| op == GIT_SHA_IGNORE_PREFIX }.last
+            return rv == version.to_s
```

Comment:
Concern: this is the opposite of the internal implementation, but given my assumptions is what _sort of_ made sense... But then, why would the requirement not just use `= ${commitSHA}` for that?

I'm leaning towards the base assumption being wrong: `!! ${commitSHA}` means: ignore versions other than `commitSHA`.
I'm not sure about the use case for that - why wouldn't you just ignore all versions?